JASP-Engine/JASP/R/multinomialtest.R

```diff
@@ -15,546 +15,447 @@
-      title <- paste0(interval, "% Confidence Interval")
-      fields[[length(fields)+1]] <- list(name="lowerCI",
-                                         title="Lower",
-                                         type = "number",
-                                         format = "sf:4;dp:3",
-                                         overTitle = title)
-      fields[[length(fields)+1]] <- list(name="upperCI",
-                                        title="Upper",
-                                        type = "number",
-                                        format = "sf:4;dp:3",
-                                        overTitle = title)
-      .addFootnote(footnotes, symbol = "<em>Note.</em>", "Confidence intervals are based on independent binomial distributions.")
-    }
-
-    # Then we fill the columns with the information
-    if (options[["countProp"]]=="descCounts"){
-      div <- 1
-    } else {
-      div <- sum(chisqResults[[1]][["observed"]])
-    }
-
-    tableFrame <- data.frame(
-      factor = names(chisqResults[[1]][["observed"]]),
-      observed = as.numeric(chisqResults[[1]][["observed"]])/div,
-      stringsAsFactors = FALSE
-    )
-
-    for (r in chisqResults){
-      tableFrame <- cbind(tableFrame, r[["expected"]]/div)
-    }
-
-    if (length(nms) == 1) {
-      colnames(tableFrame)[-(1:2)] <- "expected"
-    } else {
-      colnames(tableFrame)[-(1:2)] <- nms
-    }
-
-    # Add confidenceInterval to the tableFrame
-    if (options$confidenceInterval){
-      ciDf       <- .multComputeCIs(chisqResults[[1]][["observed"]], options$confidenceIntervalInterval, 
-                                    scale = options$countProp)
-      tableFrame <- cbind(tableFrame, ciDf)
-      if (any(is.nan(unlist(tableFrame[, c('lowerCI', 'upperCI')])))){
-        .addFootnote(footnotes, symbol = "<em>Note.</em>", "Could not compute confidence intervals.")
-      }
-    }
-
-    rows <- list()
-
-    for (i in 1:nrow(tableFrame)){
-      rows[[i]] <- as.list(tableFrame[i,])
+  }
+  if (options$confidenceInterval){
+    interval <- 100 * options$confidenceIntervalInterval
+    title <- paste0(interval, "% Confidence Interval")
+    descriptivesTable$addColumnInfo(name = "lowerCI", title = "Lower", 
+                                    type = "number", overtitle = title)
+    descriptivesTable$addColumnInfo(name = "upperCI", title = "Upper", 
+                                    type = "number", overtitle = title)
+  }
+  jaspResults[["descriptivesTable"]] <- descriptivesTable
+  if (!ready) 
+    return()
+  res <- try(.multinomDescriptivesTableFill(jaspResults, options, chisqResults))
+  .multinomialSetError(res, descriptivesTable)
+  if(options$confidenceInterval) {
+    message <- "Confidence intervals are based on independent binomial distributions."
+    descriptivesTable$addFootnote(message)
+    if (any(is.nan(unlist(descriptivesTable[["data"]][, c('lowerCI', 'upperCI')])))){
```

Comment:
```suggestion
    if (anyNA(unlist(descriptivesTable[["data"]][, c('lowerCI', 'upperCI')]))){
```
should also work and is a little shorter :)